### PR TITLE
Add id for different z-index inside SVG

### PR DESCRIPTION
**Pull request by jindrichkuba**
I need to use different z-index 

like:

    
    <circle id="one" fill="green" cx="100" cy="105" r="20" />
    
    
    <circle id="two" fill="orange" cx="100" cy="95" r="20" />
    
    
    <use xlink:href="#one"/>

My idea is using use xlink:href for Ids

### CODE REVIEW

**Comment by jindrichkuba** [comment]
Guys, If I will create similar PR for 1.x it will be merge? What is current situation with branches and prototype master, what is current supported version?

**Comment by scallaway** [comment]
> Guys, If I will create similar PR for 1.x it will be merge? What is current situation with branches and prototype master, what is current supported version?

I am wondering the same thing as I have a PR that's ready to be opened for 1.x but not sure if it's worthwhile given the current updates. Don't want to open one for it to have no chance of getting merged!

Any information on this would be great.

File: src/cartesian/Scatter.tsx [hunk 1]
```diff
@@ -219,14 +219,15 @@ class Scatter extends PureComponent<Props, State> {
...
-    const { shape, activeShape, activeIndex } = this.props;
+    const { shape, activeShape, activeIndex, id } = this.props;
     const baseProps = filterProps(this.props);
 
     return points.map((entry, i) => {
       const props = { key: `symbol-${i}`, ...baseProps, ...entry };
 
       return (
         <Layer
+          id={id && id + i}
```
Comment (xile611) [inline src/cartesian/Scatter.tsx]: the `id` maybe string.

File: src/shape/Curve.tsx [hunk 2]
```diff
@@ -111,6 +111,7 @@ class Curve extends PureComponent<Props> {
     return (
       <path
         {...filterProps(this.props)}
+        id={null}
```
Comment (xile611) [inline src/shape/Curve.tsx]: why set the  id of <Curve /> to be null?

**Comment by xile611** [comment]
@jindrichkuba can you fix the conflicts and reopen this PR ?

**Comment by jindrichkuba** [comment]
I will do that today or tomorrow, I will fix the conflict, I will let you know, thank you for code review

**Comment by xile611** [comment]
@jindrichkuba Please reopen this PR if you fix the conflicts. Thanks~

**Pull request closed without merging**